indico/modules/events/registration/controllers/api/misc.py

```diff
@@ -6,9 +6,17 @@
 # LICENSE file for more details.
 
 from indico.modules.events.controllers.base import RHProtectedEventBase
+from indico.modules.events.timetable.legacy import TimetableSerializer
 
 
 class RHAPIRegistrationForms(RHProtectedEventBase):
     def _process(self):
         from indico.modules.events.registration.schemas import RegistrationFormPrincipalSchema
         return RegistrationFormPrincipalSchema(many=True).jsonify(self.event.registration_forms)
+
+
+class RHAPIEventTimeTable(RHProtectedEventBase):
+    def _process(self):
+        timetable_data = TimetableSerializer(self.event, management=True).serialize_timetable()
```

Comment:
This contains way too much! Just return the session blocks. You also probably do not need a separate API for this but I think for now it's easier to keep it and then we can possibly optimize it later.

Reply:
I agree, but, how? neither of the methods provided by TimeTableSerializer returned only the blocks, so I decided to return the entire table and filter it in the component

Reply:
Simply don't use the timetable serializer? ;)

I haven't checked which data you need exactly, but something as simple as this probably does the job for the beginning (could change to a marshmallow schema later):

```python
>>> [{'id': sb.id, 'start_dt': sb.start_dt, 'title': sb.full_title} for s in event.sessions for sb in s.blocks]
[{'id': 426067,
  'start_dt': datetime.datetime(2024, 2, 14, 17, 0, tzinfo=<UTC>),
  'title': 's1: s1b1'},
 {'id': 426068,
  'start_dt': datetime.datetime(2024, 2, 14, 18, 0, tzinfo=<UTC>),
  'title': 's1: s1b2'},
 {'id': 426069,
  'start_dt': datetime.datetime(2024, 2, 14, 17, 0, tzinfo=<UTC>),
  'title': 's2: s2b1'},
 {'id': 426070,
  'start_dt': datetime.datetime(2024, 2, 14, 18, 0, tzinfo=<UTC>),
  'title': 's2: s2b2'}]
```

Reply:
I cannot believe I started using that approach and cancelled it when I discovered TimeTableSerializer existed and thought you were going to tell me to use it